subt/tools/validator.py

```diff
@@ -0,0 +1,138 @@
+"""
+  Validate pose3d against ground truth from ignition simulator
+"""
+import datetime
+
+from osgar.logger import LogReader, lookup_stream_id
+from osgar.lib.serialize import deserialize
+import subt.tools.ignstate as ign
+from subt.trace import distance3D
+
+
+MAX_SIMULATION_DURATION = 3700  # 1hour with a small buffer
+
+SIM_TIME_STREAM = 'rosmsg.sim_time_sec'
+ORIGIN_STREAM = 'rosmsg.origin'
+
+
+def evaluate_poses(poses, gt_poses, time_step_sec=1.0):
+    if len(poses) == 0 or len(gt_poses) == 0:
+        return []
+    time_limit = max(poses[0][0], gt_poses[0][0])
+    end_time = min(poses[-1][0], gt_poses[-1][0])
+
+    i, j = 0, 0
+    arr = []
+    while time_limit <= end_time:
+        while poses[i][0] < time_limit:
+            i += 1
+        while gt_poses[j][0] < time_limit:
+            j += 1
+        dist = distance3D(poses[i][1:], gt_poses[j][1:])
+        diff = [a - b for a, b in zip(poses[i][1:], gt_poses[j][1:])]
+        arr.append((time_limit, dist, *diff))
+        time_limit += time_step_sec
```

Comment:
I think this might not result in a correct calculation if the value of `time_limit` starts as the value of `poses[0][0]`. The worst case being the value 0.9999s where the values of poses would be mismatched by 1s. Maybe a better approach would be to take the first timestamp from `gt_poses` that is bigger than `poses[0][0]` (and call it a `start_time` for an easier understanding of the code). After that we could just use the tick from `gt_poses` (I mean for each gt_poses item find the closest poses item) thus be oblivious of the sim_time resolution.

Reply:
I didn't realize that timestamps in both arrays are `sim_time` so the trouble case cannot happen. 

So if we can have the same resolution in both arrays (the other open comment), we only need to find out where to start but then we can advance in both arrays synchronously (index += 1).